src/Microsoft.Health.Fhir.Liquid.Converter.UnitTests/Filters/DateFiltersTests.cs

```diff
@@ -24,13 +25,32 @@ public static IEnumerable<object[]> GetValidDataForAddHyphensDate()
 
         public static IEnumerable<object[]> GetValidDataForFormatAsDateTime()
         {
-            yield return new object[] { null, null };
-            yield return new object[] { string.Empty, string.Empty };
-            yield return new object[] { @"2001", @"2001" };
-            yield return new object[] { @"200101", @"2001-01" };
-            yield return new object[] { @"20050110045253", @"2005-01-10T04:52:53Z" };
-            yield return new object[] { @"20110103143428-0800", @"2011-01-03T14:34:28-08:00" };
-            yield return new object[] { @"19701231115959+0600", @"1970-12-31T11:59:59+06:00" };
+            // TimeZoneHandling does not affect dateTime without time
+            yield return new object[] { null, "preserve", null };
+            yield return new object[] { null, "utc", null };
+            yield return new object[] { null, "local", null };
+            yield return new object[] { string.Empty, "preserve", string.Empty };
+            yield return new object[] { string.Empty, "utc", string.Empty };
+            yield return new object[] { string.Empty, "local", string.Empty };
+            yield return new object[] { @"2001", "preserve", @"2001" };
+            yield return new object[] { @"2001", "utc", @"2001" };
+            yield return new object[] { @"2001", "local", @"2001" };
+            yield return new object[] { @"200101", "preserve", @"2001-01" };
+            yield return new object[] { @"200101", "utc", @"2001-01" };
+            yield return new object[] { @"200101", "local", @"2001-01" };
+
+            // If no time zone provided, it is treated as local
+            yield return new object[] { @"20050110045253", "preserve", @"2005-01-10T04:52:53" };
+            yield return new object[] { @"20050110045253", "utc", @"2005-01-09T20:52:53Z" };
+            yield return new object[] { @"20050110045253", "local", @"2005-01-10T04:52:53" };
+
+            // If time zone provided, it should be formatted according to TimeZoneHandling
+            yield return new object[] { @"20110103143428-0800", "preserve", @"2011-01-03T14:34:28-08:00" };
+            yield return new object[] { @"20110103143428-0800", "utc", @"2011-01-03T22:34:28Z" };
+            yield return new object[] { @"20110103143428-0800", "local", @"2011-01-04T06:34:28+08:00" };
```

Comment:
The local timezone is Beijing time here, will exception occur when this engine is ran in other timezone? Say the engine is packaged to nuget, and ran in pipeline in Redmond.

Reply:
Yes, the tests are running in UTC time zone, so tests for local time validation cannot pass. I'm looking into how to solve this.